06_puppy/sirigithub/mapstore.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+)
+
+type MapStore map[int]Puppy
+
+func NewMapStore() MapStore {
+	return MapStore{}
+}
+
+// CreatePuppy takes a  user provided puppy, and creates a new Puppy in the store
+// returns the ID of the new Puppy.
+func (m MapStore) CreatePuppy(puppy *Puppy) int {
+	puppy.ID = len(m)
+	m[puppy.ID] = *puppy
+	return puppy.ID
+}
+
+// UpdatePuppy overrides an existing puppy in the store,
+// returns an error if id is not found or does not match the puppy ID
```

Comment:
nit: doc strings should end with a period as they are English sentences when presented in the documentation. Please fix this throughout (all comments in all files).

See: https://blog.golang.org/godoc-documenting-go-code

Reply:
Yes, well spotted done.